test/serializers/associations_test.rb

```diff
@@ -145,6 +145,26 @@ def test_associations_custom_keys
         assert expected_association_keys.include? :writer
         assert expected_association_keys.include? :site
       end
+
+      def test_belongs_to_doesnt_load_record
+        post = Post.new({ id: 997 })
+
+        class << post
+          def blog_id
+            996
+          end
+          def blog
+            fail "should use blog_id"
+          end
+        end
+
+        post_serializer = Class.new(ActiveModel::Serializer) do
```

Comment:
You could

```ruby
post_serializer = Class.new(ActiveModel::Serializer)
Object.const_set(:PostSerializer, post_serializer)
PostSerializer.whatever
```

e.g. see https://github.com/rails-api/active_model_serializers/blob/18e5812c9310b2b0240950a4a0d92e2200aeb17c/test/serializers/adapter_for_test.rb#L93-L144